google/auth/compute_engine/_metadata.py

```diff
@@ -229,7 +231,7 @@ def get(
         raise exceptions.TransportError(
```

Comment:
Here we should consult `last_exception` and `response` to construct the right error:

(1) Non-retryable error encountered: `response.status`
(2) Ran out of retries: `last_exception`
(3) Ran out of retries: `response.status`

Reply:
Yes, this was the first iteration of the idea. It seems to me that converting response to an exception works just as well and simplifies the logic since only the last exception needs to be tracked. The logging etc does not change. Do you want us to change this?

Reply:
I think the code flow is more confusing with exceptions, consistent with the [Python style guide](go/pystyle#exceptions), so I'd like it changed.  Happy to get a 2nd opinion from someone with more Python experience like @sai-sunder-s or @parthea 

/cc @Ark-kun